frontend/packages/kubevirt-plugin/src/components/modals/nic-modal/nic-modal.tsx

```diff
@@ -257,15 +257,20 @@ export const NICModal = withHandlePromise((props: NICModalProps) => {
               isDisabled={isDisabled('model') || interfaceType === NetworkInterfaceType.SRIOV}
             >
               <FormSelectPlaceholderOption isDisabled placeholder="--- Select Model ---" />
-              {NetworkInterfaceModel.getAll().map((ifaceModel) => {
-                return (
-                  <FormSelectOption
-                    key={ifaceModel.getValue()}
-                    value={ifaceModel.getValue()}
-                    label={ifaceModel.toString()}
-                  />
-                );
-              })}
+              {NetworkInterfaceModel.getAll()
+                .filter(
+                  (ifaceModel) =>
+                    ifaceModel.isSupported() || ifaceModel.toString() === model.toString(),
```

Comment:
```suggestion
                    ifaceModel.isSupported() || ifaceModel === model,
```

you can just compare - plus no NPE will occur